security/pkg/pki/util/generate_cert.go

```diff
@@ -204,25 +208,32 @@ func genCertTemplateFromOptions(options CertOptions) (*x509.Certificate, error)
 		return nil, err
 	}
 
+	subject := pkix.Name{
+		Organization: []string{options.Org},
+	}
+
 	exts := []pkix.Extension{}
 	if h := options.Host; len(h) > 0 {
 		s, err := BuildSubjectAltNameExtension(h)
 		if err != nil {
 			return nil, err
 		}
+		if options.IsDualUse {
+			// only first hostname used for CN
+			first := strings.SplitN(h, ",", 2)[0]
+			subject.CommonName = first
```

Comment:
A minor thing: CN has a length limitation of 64 chars. Can we put a length check here and if the value is longer than that, give up and log an error? Thanks.

Reply:
Give up on the CN, or give up the entire cert?

Reply:
I think we can give up on setting the CN.